lib/src/header_parser/sub_parsers/macro_parser.dart

```diff
@@ -66,7 +66,13 @@ List<Constant>? parseSavedMacros() {
   final index = clang.clang_createIndex(0, 0);
   Pointer<Pointer<Utf8>> clangCmdArgs = nullptr;
   var cmdLen = 0;
-  clangCmdArgs = createDynamicStringArray(config.compilerOpts);
+
+  final compilerOpts = config.compilerOpts;
+  config.headers.entryPoints.followedBy([file.path]).forEach((entryFile) {
+    compilerOpts.add("-imacros$entryFile");
```

Comment:
Is it necessary to add the `-imacros` flag to compiler-opts?

Reply:
Yes. The compiler doesn't properly link the macros between entry and generated files otherwise.